docker/server/entrypoint.sh

```diff
@@ -109,7 +109,7 @@ if [ -n "$CLICKHOUSE_USER" ] && [ "$CLICKHOUSE_USER" != "default" ] || [ -n "$CL
           <networks>
             <ip>::/0</ip>
           </networks>
-          <password>${CLICKHOUSE_PASSWORD}</password>
+          <password><![CDATA[${CLICKHOUSE_PASSWORD//]]>/]]]]><![CDATA[>}]]></password>
```

Comment:
I think the issue is the substitution of `${CLICKHOUSE_PASSWORD}` won't happen when `default-user.xml` is written, which is problematic.

Reply:
`default-user.xml` will be overwritten by the entrypoint. The condition doesn't check an existence of this file

Reply:
I am not talking about whether `default-user.xml` will be written or not. I am talking about the substituion of `${CLICKHOUSE_PASSWORD}`. What I mean is previously in the `default-user.xml` there was a line like this:
```
<password>MyVerySecretPassword42</password>
```
This is because `${CLICKHOUSE_PASSWORD}` got substituted with its value. Now we will have have this:
```
<![CDATA[${CLICKHOUSE_PASSWORD//]]>/]]]]><![CDATA[>}]]>
```
This means `${CLICKHOUSE_PASSWORD}` won't substitued when `default-user.xml` is written, thus ClickHouse probably will be misconfigured with a wrong password, so the password will be incorrect.

Do I miss something here?

Reply:
And now this line would be substituted like this:
```
<password><![CDATA[MyVerySecretPassword42]]></password>
```

Which is pretty the same in terms of XML. I checked this hack on my environment and it works as expected. 

Why do you think it won't be substituted? `${A//from/to}` is a valid bash syntax, which replaces all occurrences of `from` with `to`

Reply:
Ah, okay then, I learnt something new today! I didn't know about `${A//from/to}` mean anything special in bash. Sorry for my lack of knowledge and thanks for the contribution! I will check this then locally to make sure it works for me too. I trust you that it works on your machine, but slightly different environments might behave differently.